pkg/service/manifest/manifest.go

```diff
@@ -65,34 +74,129 @@ func NewManifestService(config config.ManifestServiceConfig, s storage.ServiceSt
 		storage:    manifestStorage,
 		config:     config,
 		keyStore:   keyStore,
+		resolver:   didResolver,
 		credential: credential,
 	}, nil
 }
 
 func (s Service) CreateManifest(request CreateManifestRequest) (*CreateManifestResponse, error) {
+
 	logrus.Debugf("creating manifest: %+v", request)
 
+	// validate the request
 	m := request.Manifest
 	if err := m.IsValid(); err != nil {
 		return nil, util.LoggingErrorMsg(err, "manifest is not valid")
 	}
 
+	// sign the manifest
+	manifestJWT, err := s.signManifestJWT(m)
+	if err != nil {
+		return nil, util.LoggingErrorMsg(err, "could not sign manifest")
+	}
+
 	// store the manifest
 	storageRequest := manifeststorage.StoredManifest{
-		ID:       m.ID,
-		Manifest: m,
-		Issuer:   m.Issuer.ID,
+		ID:          m.ID,
+		Issuer:      m.Issuer.ID,
+		Manifest:    m,
+		ManifestJWT: *manifestJWT,
 	}
 
-	if err := s.storage.StoreManifest(storageRequest); err != nil {
+	if err = s.storage.StoreManifest(storageRequest); err != nil {
 		return nil, util.LoggingErrorMsg(err, "could not store manifest")
 	}
 
 	// return the result
-	response := CreateManifestResponse{Manifest: m}
+	response := CreateManifestResponse{Manifest: m, ManifestJWT: *manifestJWT}
 	return &response, nil
 }
 
+func (s Service) signManifestJWT(m manifest.CredentialManifest) (*keyaccess.JWT, error) {
+	issuerID := m.Issuer.ID
+	gotKey, err := s.keyStore.GetKey(keystore.GetKeyRequest{ID: issuerID})
+	if err != nil {
+		errMsg := fmt.Sprintf("could not get key for signing manifest with key<%s>", issuerID)
+		return nil, util.LoggingErrorMsg(err, errMsg)
```

Comment:
errors.wrap vs util.LoggingErrorMsg(err, errMsg)
Eventually we will need to go through code and standardize, I'm guilty of this too we have 5 different ways to return errors

Reply:
good catch. yeah, we need to have some rules on where we log vs where we return errors

definitely on the caller site but maybe we should be logging in each layer if there are more specific errors. like, in storage log specific storage errors, in the app layer log generation/modification errors, service layer something more generic...